packages/editor/src/store/selectors.js

```diff
@@ -360,6 +361,20 @@ export function isEditedPostSaveable( state ) {
 	);
 }
 
+/**
+ * Determines whether all the block available in post are known or registered block or not.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {boolean} whether all the block available in post are known block or not.
+ */
+export function hasKnownBlocks( state ) {
+	return getBlockOrder( state ).some( ( clientId ) => (
+		state.editor.present.blocksByClientId[ clientId ].name !== getFreeformContentHandlerName() &&
+		state.editor.present.blocksByClientId[ clientId ].name !== getUnregisteredTypeHandlerName()
```

Comment:
The split of these two functions has me wondering if the semantics are meaningfully different enough to consider. "Unknown" would be _truly_ a block, with demarcations, and thus non-empty for save. I think what we're really targeting is specifically freeform content.

Thus, I think we don't need this line and it could be dropped.

In doing so, and in reconsideration of targeting specifically freeform content, I wonder if the function name we've chosen `hasKnownBlocks` is very accurate. I'll need to sit and think on this one...

Reply:
> I wonder if the function name we've chosen hasKnownBlocks is very accurate. I'll need to sit and think on this one...

If we are just going to check that the post have any freeform block or not then `hasFreeFromBlock` name will be more suitable for this.

Reply:
That name seems to suit the purpose, but it highlights to me that it's _so specific_ to the `isEditedPostEmpty` use-case that we might be better off to include the logic there, rather than to expose a new separate selector.

Thinking something like (haven't verified the logic):

```js
export function isEditedPostEmpty( state ) {
	const blocks = getBlocksForSerialization( state );
	if ( ! blocks.length ) {
		return true;
	}

	const freeformBlockName = getFreeformContentHandlerName();
	if ( ! some( blocks, { name: freeformBlockName } ) ) {
		return false;
	}

	// Content serialization is considered an expensive operation, and should
	// only be considered after more trivial operations of assuming presence of
	// non-freeform blocks as implying that serializable content exists.
	return ! getEditedPostAttribute( state, 'content' );
}
```

Reply:
@aduth This works correctly.

Reply:
@rahulsprajapati @aduth : You, the code works correctly, but while writing test cases for this, existing test case `should return false if edits include a non-empty content property` fails. So I think need to check for a more suitable solution. What do you guys say?